tardis/workflows/tardis_he_workflow.py

```diff
@@ -0,0 +1,196 @@
+from tardis.energy_input.gamma_ray_channel import (
+    create_isotope_dicts,
+    create_inventories_dict,
+    calculate_total_decays,
+    create_isotope_decay_df,
+    time_evolve_cumulative_decay,
+)
+from tardis.energy_input.main_gamma_ray_loop import run_gamma_ray_loop, get_effective_time_array
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class TARDISHEWorkflow:
+    def __init__(self, atom_data, configuration, config_type="yaml"):
+
+        if config_type == "csvy":
+            self.simulation_state = SimulationState.from_csvy(
+                configuration, atom_data
+            )
+        else:
+            self.simulation_state = SimulationState.from_config(
+                configuration, atom_data
+            )
+
+        self.gamma_ray_lines = atom_data.decay_radiation_data
+
+        self.shell_masses = (
+            self.simulation_state.volume * self.simulation_state.density
+        )
+
+        self.isotopic_mass_fraction = (
+            self.simulation_state.composition.isotopic_mass_fraction
+        )
+
+    def calculate_total_decays(self, time_start, time_end):
+        """
+        Calculate the total number of decays for each isotope in the simulation
+        state between t_start and t_end for each shell.
+
+        Parameters
+        ----------
+        time_start : float
+            The start time in days.
+        time_end : float
+            The end time in days.
+
+        Returns
+        -------
+        total_decays : dataframe
+            A dataframe containing the total number of decays for each isotope
+            in the simulation state between t_start and t_end for each shell.
+        """
+        isotopes = create_isotope_dicts(
+            self.isotopic_mass_fraction, self.shell_masses
+        )
+        inventories = create_inventories_dict(isotopes)
+
+        total_decays = calculate_total_decays(
+            inventories, time_end - time_start
+        )
+
+        return total_decays
+    
+    def decay_isotopes_expanded(self, total_decays):
+        """
+        Expand the total decays dataframe to include the decay energy for each gamma-X ray
+        transition in each shell between t_start and t_end.
+
+        Parameters
+        ----------
+        total_decays : dataframe
+            A dataframe containing the total number of decays for each isotope
+            in the simulation state between t_start and t_end for each shell.
+        
+        Returns
+        -------
+        decay_isotopes : dataframe
+            A dataframe containing the total number of decays for each isotope
+            in the simulation state between t_start and t_end for each shell
+            expanded to include the decay energy for each gamma-X ray transition.
+        """
+        return create_isotope_decay_df(total_decays, self.gamma_ray_lines)
+    
+    def time_evolve_cumulative_decay_expanded(self, times):
```

Comment:
Consider merging this method with the `get_times` method since both use time as arguments.